packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -1969,38 +1969,44 @@ addEventListener('fetch', event => {});`
 			mockKeyListRequest(kvNamespace.id, []);
 			mockUploadAssetsToKVRequest(kvNamespace.id, assets);
 
-			await runWrangler("deploy --assets assets --latest --name test-name");
+			await runWrangler(
+				"deploy --legacy-assets assets --latest --name test-name"
+			);
 
 			expect(std).toMatchInlineSnapshot(`
-			Object {
-			  "debug": "",
-			  "err": "",
-			  "info": "Fetching list of already uploaded assets...
-			Building list of assets to upload...
-			 + file-1.2ca234f380.txt (uploading new version of file-1.txt)
-			 + file-2.5938485188.txt (uploading new version of file-2.txt)
-			Uploading 2 new assets...
-			Uploaded 100% [2 out of 2]",
-			  "out": "↗️  Done syncing assets
-			Total Upload: xx KiB / gzip: xx KiB
-			Uploaded test-name (TIMINGS)
-			Published test-name (TIMINGS)
-			  https://test-name.test-sub-domain.workers.dev
-			Current Deployment ID: Galaxy-Class
-			Current Version ID: Galaxy-Class
+				Object {
+				  "debug": "",
+				  "err": "",
+				  "info": "Fetching list of already uploaded assets...
+				Building list of assets to upload...
+				 + file-1.2ca234f380.txt (uploading new version of file-1.txt)
+				 + file-2.5938485188.txt (uploading new version of file-2.txt)
+				Uploading 2 new assets...
+				Uploaded 100% [2 out of 2]",
+				  "out": "↗️  Done syncing assets
+				Total Upload: xx KiB / gzip: xx KiB
+				Uploaded test-name (TIMINGS)
+				Published test-name (TIMINGS)
+				  https://test-name.test-sub-domain.workers.dev
+				Current Deployment ID: Galaxy-Class
+				Current Version ID: Galaxy-Class
 
 
-			Note: Deployment ID has been renamed to Version ID. Deployment ID is present to maintain compatibility with the previous behavior of this command. This output will change in a future version of Wrangler. To learn more visit: https://developers.cloudflare.com/workers/configuration/versions-and-deployments",
-			  "warn": "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mThe --assets argument is experimental and may change or break at any time[0m
+				Note: Deployment ID has been renamed to Version ID. Deployment ID is present to maintain compatibility with the previous behavior of this command. This output will change in a future version of Wrangler. To learn more visit: https://developers.cloudflare.com/workers/configuration/versions-and-deployments",
+				  "warn": "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mThe behavior of the experimental --assets command will be changing on August 15th.[0m
```

Comment:
Similarly to the config, if the user actually uses `--legacy-assets` then it doesn't really make sense to give this warning, no?